elser-ingest-speedtest/challenges/Multi-Doc-Multi-Parametric-Bulk-Size-Search.json

```diff
@@ -0,0 +1,127 @@
+{
+  "name": "ELSER-Size-Search-Multi-Doc-Multi-Parametric-Ingest",
+  "description": "Test the throughput of ELSER in various scenarios",
+  "default": false,
+  "schedule": [
+    {%- set model_id = model_id | default('default') %}
+    {
+      "name": "put elser v2 platform specific",
+      "operation": {
+        "operation-type": "put-elser",
+        "include-in-reporting": false,
+        "model_id": "{{ model_id }}"
+      }
+    },
+    {
+      "name": "setup-elser-pipeline",
+      "operation": {
+        "operation-type": "put-pipeline",
+        "id": "{{ pipeline_name | default('default-pipeline') }}",
+        "body": {
+          "description": "Pipes documents to ELSER",
+          "processors": [
+            {
+              "inference": {
+                "model_id": "{{ model_id }}",
+                "target_field": "ml",
+                "field_map": {
+                  "body": "text_field"
+                },
+                "inference_config": {
+                  "text_expansion": {
+                    "results_field": "tokens"
+                  }
+                }
+              }
+            }
+          ],
+          "on_failure": [
+            {
+              "set": {
+                "field": "_index",
+                {% raw %}
+                "description": "Record error information",
+                "field": "error_information",
+                "value": "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message {{ _ingest.on_failure_message }}"
+                {% endraw %}
+              }
+            }
+          ]
+        }
+      }
+    },
+    {
+      "operation": "delete-index",
+      "name": "first-delete-index"
+    },
+    {%- set p_max_number_of_allocations = max_number_of_allocations | default(1) %}
+    {%- set p_min_number_of_allocations = min_number_of_allocations | default(1) %}
+    {%- set p_min_doc_size = min_doc_size | default(128) %}
+    {%- set p_max_doc_size = max_doc_size | default(128) %}
+    {%- set p_doc_size_interval = doc_size_interval | default(128) %}
+
+    {%- for doc_size in range(p_min_doc_size, p_max_doc_size + 1, p_doc_size_interval) %}
+
+    {%- set bulk_size = bulk_size | default(100) %}
+    {%- for bulk_size_multiplier in range(1, 17, 2) %}
+    {%- set bulk_size_mult_int = (bulk_size_multiplier/10)*(max_doc_size/doc_size) | round %}
```

Comment:
The readme says the bulk size is scaled by
```
bulk_size*(max_doc_size/doc_size)
```

Reply:
the bulk size scaling changed a few times after I wrote that readme. I've updated the readme with the new formula

Reply:
also, this challenge isn't mentioned in the readme at all because it isn't really intended for people to run as a benchmark, but rather allows people setting up benchmarks to determine good scaling values.